### PR TITLE
typescript: use DOM.Iterable type library

### PR DESCRIPTION
**Pull request by lightmare**
Enable `dom.iterable` in `tsconfig` and remove a few unnecessary `ts-ignore` comments, `as Iterable<Element>` assertions, and `Array.from` calls.

### CODE REVIEW

File: src/header-parser.ts [hunk 1]
```diff
@@ -321,15 +317,13 @@ export function formatPreamble(
     }
   }
   if (description != null) {
-    // @ts-ignore childNodes is iterable
-    const isJustElements = [...(description.childNodes as Iterable<Node>)].every(
+    const isJustElements = Array.prototype.every.call(
```
Comment (bakkot) [inline src/header-parser.ts:320]: Can you revert this change? As a matter of style I prefer spreading and calling array methods directly over using `.call`.

File: src/Spec.ts [hunk 2]
```diff
@@ -1243,10 +1236,13 @@ ${await utils.readFile(path.join(__dirname, '../js/multipage.js'))}
         });
         continue outer;
       }
+      if (!header) {
```
Comment (bakkot) [inline src/Spec.ts:1239]: This is unreachable; please remove it.

File: src/Spec.ts [hunk 3]
```diff
@@ -968,20 +967,20 @@ ${await utils.readFile(path.join(__dirname, '../js/multipage.js'))}
...
-    await this.loadBiblio(path.join(__dirname, '../ecma262biblio.json'));
+    await this.loadBiblio(__dirname, '../ecma262biblio.json');
   }
 
   private async loadBiblios() {
     this.cancellationToken.throwIfCancellationRequested();
     await Promise.all(
-      Array.from(this.doc.querySelectorAll('emu-biblio')).map(biblio =>
-        this.loadBiblio(path.join(this.rootDir, biblio.getAttribute('href')!))
+      Array.prototype.map.call(this.doc.querySelectorAll('emu-biblio'), biblio =>
```
Comment (bakkot) [inline src/Spec.ts:976]: As below, please revert this change.

File: src/Spec.ts [hunk 4]
```diff
@@ -968,20 +967,20 @@ ${await utils.readFile(path.join(__dirname, '../js/multipage.js'))}
...
+        this.loadBiblio(this.rootDir, biblio.getAttribute('href')!)
       )
     );
   }
 
-  private async loadBiblio(file: string) {
-    const contents = await this.fetch(file);
+  private async loadBiblio(...file: string[]) {
```
Comment (bakkot) [inline src/Spec.ts:982]: Please revert this change; it's not related, and I prefer the original.

File: src/header-parser.ts (hunk 1 above)
Comment (lightmare) [reply src/header-parser.ts:320]: That's forfeiting the possibility of not exhausting the whole list if every short-circuits; but okay, it wouldn't really buy anything here, anyway.

**Comment by lightmare** [comment]
Amended with loadBiblio and Array.prototype.every reverted.

**Review by bakkot — APPROVED** [review]

**Comment by bakkot** [comment]
Thanks!
